mission_control/test/test_altitude_heading_behavior.py

```diff
@@ -0,0 +1,123 @@
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+"""
+import sys
+import os
+import unittest
+import math
+import rosnode
+import rospy
+import rostest
+from mission_control.msg import ReportExecuteMissionState
+from mission_control.msg import AltitudeHeading
+from auv_interfaces.msg import StateStamped
+from mission_interface import MissionInterface
+from mission_interface import wait_for
+
+
+class TestAltitudeHeadingBehavior(unittest.TestCase):
+    """ 
+        The test loads and execute an altitude heading behavior.
+        -   Load the mission altitude_heading_mission_test.xml
+        -   Execute the mission
+        -   Check if the behavior publishes the goal
+        -   Simulate auv_inteface data to finish the behavior
+        -   Test if the mission is COMPLETE
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        rospy.init_node('test_altitude_heading_behavior')
+
+    def setUp(self):
+        self.altitude_heading_goal = AltitudeHeading()
+        self.mission = MissionInterface()
+
+        self.altitude_heading_msg = rospy.Subscriber(
+            '/mngr/altitude_heading',
+            AltitudeHeading,
+            self.altitude_heading_goal_callback)
+
+        self.simulated_auv_interface_data_pub = rospy.Publisher(
+            '/state',
+            StateStamped,
+            queue_size=1)
+
+    def altitude_heading_goal_callback(self, msg):
+        self.altitude_heading_goal = msg
+        rospy.loginfo(msg)
+
+    def test_mission_with_altitude_heading_behavior(self):
+        self.mission.load_mission('altitude_heading_mission_test.xml')
+        self.mission.execute_mission()
+        self.mission.read_behavior_parameters('AltitudeHeadingBehavior')
+        altitude = self.mission.get_behavior_parameter('altitude')
+        heading = self.mission.get_behavior_parameter('heading')
+        speed_knots = self.mission.get_behavior_parameter('speed_knots')
+        enable_mask = 0
+
+        # Calculate the mask
+        def get_mask(value, mask):
+            return mask if value else 0
+        enable_mask |= get_mask(altitude, AltitudeHeading.ALTITUDE_ENA)
```

Comment:
@patapufate hmm, this will cause the mask to be disabled if value is 0 (or coercible to False). Consider:
```suggestion
        if altitude is not None:
            enable_mask |= AltitudeHeading.ALTITUDE_ENA
```